--- a/models/yolo.py
+++ b/models/yolo.py
@@ -23,6 +23,7 @@ except ImportError:
 class Detect(nn.Module):
     stride = None  # strides computed during build
     export = False  # onnx export
+    ignore_detect_layer = False
     end2end = False
     include_nms = False
     concat = False
@@ -43,6 +44,13 @@ class Detect(nn.Module):
         # x = x.copy()  # for profiling
         z = []  # inference output
         self.training |= self.export
+        if self.ignore_detect_layer:
+            for i in range(self.nl):
+                x[i] = self.m[i](x[i])
+                y = x[i].sigmoid()
+                # print("Detect forward ignore_detect_layer: {} i:{}".format(self.ignore_detect_layer,i))
+                z.append(y)
+            return tuple(z)
         for i in range(self.nl):
             x[i] = self.m[i](x[i])  # conv
             bs, _, ny, nx = x[i].shape  # x(bs,255,20,20) to x(bs,3,20,20,85)
